CMakeLists.txt

```diff
@@ -69,17 +69,20 @@ tr_auto_option(WITH_SYSTEMD         "Add support for systemd startup notificatio
 
 set(TR_NAME ${PROJECT_NAME})
 
-# convention: -TR MAJOR MINOR MAINT STATUS - (each a single char)
-# STATUS: "X" for prerelease beta builds,
-#         "Z" for unsupported trunk builds,
-#         "0" for stable, supported releases
+# convention: -TR MAJOR MINOR PATCH TYPE - (each a single char)
+# MAJOR: https://semver.org/ major in https://en.wikipedia.org/wiki/Base62
+# MINOR: https://semver.org/ minor in https://en.wikipedia.org/wiki/Base62
+# PATCH: https://semver.org/ patch in https://en.wikipedia.org/wiki/Base62
+# TYPE: '0' for stable releases, X' for beta releases, 'Z' for dev builds
+#
 # these should be the only two lines you need to change
-set(TR_USER_AGENT_PREFIX "3.00+")
+set(TR_USER_AGENT_PREFIX "3.0.0+")
```

Comment:
Based on clients.cc changes, should this now also be "3.0.0 (Dev)"? If so, `CPACK_SOURCE_PACKAGE_FILE_NAME` (below) and `MSI_FILENAME_VERSION` (dist/msi/CMakeLists.txt) may need to be adjusted as well.

Reply:
I had only intended the ` (Dev)` suffix for display in-app when identifying peers, but I'm open to suggestions. Do you want the filenames to be different?

Reply:
I was mainly thinking of fixing up filenames if `TR_USER_AGENT_PREFIX` changes, not as the primary goal. The actual issue to me appears to be a discrepancy in how clients will display peer versions (in torrent details dialog) vs. their own version (in about dialog), which ideally should be the same to avoid questions like "how is 4.0.0+ different from 4.0.0 (Dev) / 4.0.0 (Beta)?"

Reply:
That's a good point. @mikedld what if we go back to `+` instead of ` (Dev)`?

Reply:
That would be an acceptable fix as well, if you're okay with it :) I kinda like the idea of those suffixes though, since "+" to me always felt like something orthogonal to the release cycle, i.e. "4.0.0+ (Dev)" seems like a perfectly valid version to have if we're to indicate that it's currently in development and includes changes made after 4.0.0. Maybe I'm thinking too much about it, as there's no way of deserializing that "+" from peer ID (it's just not there)...

Reply:
Hmm `4.0.0+` [isn't legal semver](https://semver.org/#spec-item-10) anyway since semver uses `+` to denote the beginning of build metadata identifiers which must not be empty.

@mikedld I've updated the generation of `version.h` (both via CMake and via the bash script from Xcode) to use a valid semver string for the user-agent, and updated CPACK_SOURCE_PACKAGE_FILE_NAME.

However this was a pretty intrusive patch and we're close to the 4.0.0-beta.1 deadline so I'd appreciate a re-review (or even fixes, if they are needed and you're available)